staging/src/k8s.io/kubectl/pkg/cmd/events/convert_printer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"io"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+// corev1.Event type does not include GVK which is mandatory for
```

Comment:
I understand this makes the output work, but it seems like it shouldn't be this way.  I wonder why `corev1.Event` has no GVK?  Does this indicate a problem elsewhere that it is not being populated?

Reply:
Thanks for raising that issue @brianpursley.

As you might have already seen, I started a thread for that discussion in https://kubernetes.slack.com/archives/C0EG7JC6T/p1653897750585719.

But in any case, for the backwards compatibility, we need to handle cases where GVK is unset for event resource. This wrapper solves that problem but maybe we can remove this wrapper and instead we can explicitly set GVK in kubectl if it is not set?, this would also work.

Reply:
NVM, somehow `kubectl get events` handles that situation and I'll check how it does that and use the method in that PR.